libs/wire-api/src/Wire/API/Team/Feature.hs

```diff
@@ -212,60 +222,48 @@ newtype TeamFeatureStatusNoConfig = TeamFeatureStatusNoConfig
   { tfwoStatus :: TeamFeatureStatusValue
   }
   deriving newtype (Eq, Show, Generic, Typeable, Arbitrary)
+  deriving (ToJSON, FromJSON, S.ToSchema) via (Schema TeamFeatureStatusNoConfig)
 
+-- TODO: Delete
 modelTeamFeatureStatusNoConfig :: Doc.Model
 modelTeamFeatureStatusNoConfig = Doc.defineModel "TeamFeatureStatusNoConfig" $ do
-  Doc.description $ "Configuration for a team feature that has no configuration"
+  Doc.description "Configuration for a team feature that has no configuration"
   Doc.property "status" typeTeamFeatureStatusValue $ Doc.description "status"
 
-declareNamedSchemaFeatureNoConfig :: f -> Declare (Definitions Schema) NamedSchema
-declareNamedSchemaFeatureNoConfig _ =
-  pure $
-    NamedSchema (Just "TeamFeatureStatus") $
-      mempty
-        & properties .~ (fromList [("status", Inline statusValue)])
-        & required .~ ["status"]
-        & type_ ?~ SwaggerObject
-        & description ?~ "whether a given team feature is enabled"
-  where
-    statusValue =
-      mempty
-        & enum_ ?~ [String "enabled", String "disabled"]
-
 instance ToSchema TeamFeatureStatusNoConfig where
-  declareNamedSchema = declareNamedSchemaFeatureNoConfig
-
-instance FromJSON TeamFeatureStatusNoConfig where
-  parseJSON = withObject "TeamFeatureStatus" $ \ob ->
-    TeamFeatureStatusNoConfig <$> ob .: "status"
-
-instance ToJSON TeamFeatureStatusNoConfig where
-  toJSON (TeamFeatureStatusNoConfig status) = object ["status" .= status]
+  schema =
+    object "TeamFeatureStatusNoConfig" $
+      TeamFeatureStatusNoConfig
+        <$> tfwoStatus .= field "status" schema
 
 ----------------------------------------------------------------------
 -- TeamFeatureStatusWithConfig
 
+-- | FUTUREWORK: Does it ever happen that there is a config for a disabled
+-- feature? If not this type should be made isomorphic to 'Maybe cfg'.
```

Comment:
good catch, but:

```suggestion
-- | The support for disabled features with configs is intentional:
-- for instance, we want to be able to keep the config of a feature
-- that is turned on and off occasionally, and so not force the admin
-- to recreate the config every time it's turned on.
```

Reply:
@fisx this can be dealt with by writing a lenient parser which just ignores the config. Does our code ever need to access the config why a feature is disabled?

Reply:
@fisx's argument makes sense. @akshaymankar , if you are OK with it, i.e., to keep the configuration in the config file when the feature is disabled, I would accept the code suggestion by Matthias.